openapi_python_client/parser/openapi.py

```diff
@@ -263,12 +263,11 @@ def _add_parameters(
                 if duplicate.python_name == prop.python_name:  # Existing should be converted too for consistency
                     duplicate.set_python_name(f"{duplicate.python_name}_{duplicate_location}", config=config)
                 prop.set_python_name(f"{prop.python_name}_{param.param_in}", config=config)
-            else:
-                used_python_names[prop.python_name] = (prop, param.param_in)
-
-            endpoint.relative_imports.update(prop.get_imports(prefix="..."))
 
             if param.param_in == oai.ParameterLocation.QUERY:
+                # There is no NULL for query params, so nullable and not required are the same.
+                if prop.nullable or not prop.required:
```

Comment:
Ohh, this is very interesting. We originally tried to change the type annotations to include `None` (which was a mess), but I see you're proposing handling this at the parsing level. cc @bowenwr @GitOnUp